base-template-symlink.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+# Base Git hook template from https://github.com/rycus86/githooks
+#
+# It allows you to have a .githooks folder per-project that contains
+# its hooks to execute on various Git triggers.
+#
+# Version: 9912.310000-000000
+
+INSTALL_DIR=$(git config --global githooks.installDir)
+GITHOOKS_SCRIPT="$INSTALL_DIR/release/base-template.sh"
```

Comment:
Would it make sense to just store the full path to `base-template.sh` in a git config variable instead of deriving this from the `installDir`? That would prevent the need for hardcoding this filename and path here (i.e. the path within the git dir) and simplify the symlink script (and simpler means less likely to need changes in the future, which is good for a script that has duplicates spread around the filesystem).

Reply:
The script location is a derived value, for now at least, I don't it's necessary to introduce this duplication in the config yet. I'd do this when there's a need for it from the implementation.

Reply:
But the current approach has a lot more duplication, each installed version of the "symlink" script will now have the `release/base-template.sh` path hardcoded, in a place that is not intended to be updated normally (right?).

To make my suggestion more specific, I think it could be as simple as this:

```
RUNNER=$(git config --global githooks.runner)

if [ ! -x "$RUNNER" ]; then
    echo "! Githooks runner ('$RUNNER' from git config 'githooks.runner') is missing or broken"
    echo "  Please run the Githooks install script again to fix it." >&2
    exit 1
fi

# shellcheck disable=SC1090
exec "$GITHOOKS_SCRIPT" "$0" "$@"
```

I also removed the `sh` and added `exec` as suggested elsewhere, and removed the `$INSTALL_DIR` (assuming base-template will derive that from `$0`). The shellcheck thing still needs to be fixed.

You could simplify even more and just have a single line script:

```
exec "$(git config --global githooks.runner)" "$0" "$@"
```

Which would rely on the shell to generate the error message for a missing or broken runner, but that might be a bit less friendly.

Reply:
Very good, jeah I think, this is nice

Reply:
I wouldnt use `exec`for the first tries, because had troubles with the tests. I would leave that to the end

Reply:
Ive we move to a `githooks.sh` (runner) and a `githooks-dispatch.sh` the above approach would be good I think. Keep it simple stupid.

Reply:
This is open, to implement. Will probably break tests again. Keep it open to have a look at it.

Reply:
Done!